api/src/main/java/com/spectralogic/tpfr/api/ServerServiceFactoryImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.spectralogic.tpfr.api;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.logging.HttpLoggingInterceptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import retrofit2.Retrofit;
+import retrofit2.converter.simplexml.SimpleXmlConverterFactory;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static okhttp3.OkHttpClient.Builder;
+
+public class ServerServiceFactoryImpl implements ServerServiceFactory {
+
+    private final Logger LOG = LoggerFactory.getLogger(ServerServiceFactoryImpl.class);
+    private ServerService serverService;
+
+    @Override
+    public Optional<ServerService> getServerService() {
+        return Optional.ofNullable(serverService);
+    }
+
+    @Override
+    public ServerService createServerService(final String endpoint) {
+
+        LOG.info("Creating a server service to: " + endpoint);
+
+        final Builder httpClient = new Builder();
+        httpClient.addInterceptor(chain -> {
+            final Request original = chain.request();
+
+            final Request request = original.newBuilder()
+                    .header("Content-Type", "application/xml")
+                    .method(original.method(), original.body())
+                    .build();
+            try {
+                return chain.proceed(request);
+            } catch (final Exception e) {
+                LOG.error("Failed to connect to the server...");
```

Comment:
Make sure to include the exception in the error log statement at the end.  So `LOG.error("message", e)`